test/unittest/tests/block/test_extent_list_insert_block.cpp

```diff
@@ -0,0 +1,383 @@
+            }},
+          {off_size(4096 - 64, 64), // [4,032, 4,095] Just below First, Merge with First start.
+            {
+              off_size(4096 - 64, 4096 + 64), // [4,032, 8,191], First'.
+              off_size(3 * 4096, 4096),       // [12,288, 16,383] Second.
+              off_size(5 * 4096, 4096),       // [20,480, 24,575] Third.
+            }},
+          {off_size(2 * 4096, 64), // [8,192, 8,255] Just above First', Merge with First' end.
+            {
+              off_size(4096 - 64, 4096 + 128), // [4,032, 8,255], First''.
+              off_size(3 * 4096, 4096),        // [12,288, 16,383] Second.
+              off_size(5 * 4096, 4096),        // [20,480, 24,575] Third.
+            }},
+          {off_size(2 * 4096 + 64,
+             4096 - 64), // [8,256, 12,287] Just above First'', Merge First'' and Second.
+            {
+              off_size(4096 - 64, 3 * 4096 + 64), // [4,032, 16,383], First'''.
+              off_size(5 * 4096, 4096),           // [20,480, 24,575] Third.
+            }},
+          {off_size(
+             6 * 4096, 64), // [20,480, 12,287] Just above First''', Merge First''' and Third.
+            {
+              off_size(4096 - 64, 3 * 4096 + 64), // [4,032, 16,383], First'''.
+              off_size(5 * 4096, 4096 + 64),      // [20,480, 24,639] Third'.
+            }},
+        };
+
+        /* Setup. */
+        WT_EXTLIST extlist = {};
+        extlist.name = const_cast<char *>("__block_merge");
+
+        WT_BLOCK block = {};
+        block.name = "__block_merge";
+        block.allocsize = 1024;
+        block.size = 4096; // Description information.
+
+        /* Test. */
+        /* Insert/merge extents and verify. */
+        int idx = 0;
+        for (const off_size_expected &test : test_list) {
+            /* Call. */
+            REQUIRE(__ut_block_merge(session, &block, &extlist, test.test_off_size.off,
+                      test.test_off_size.size) == 0);
+            INFO("After " << idx << ". Insert/merge: " << &test.test_off_size);
+
+            utils::extlist_print_off(extlist);
+
+            /* Verify. */
+            verify_off_extent_list(extlist, test.expected_list, true);
+            ++idx;
+        }
+
+        /* Cleanup. */
+        extlist_free(session, extlist);
+    }
+}
+
+/*!
+ * A test (off) and the expected value (expected_list) for operations that need an off to modify a
+ * WT_EXTLIST.
+ */
+struct off_expected {
+    wt_off_t off;
+    std::vector<off_size> expected_list;
+};
+
+TEST_CASE("Extent Lists: block_off_remove", "[extent_list]")
+{
+    /* Build Mock session, this will automatically create a mock connection. */
+    std::shared_ptr<MockSession> mock_session = MockSession::buildTestMockSession();
+    WT_SESSION_IMPL *session = mock_session->getWtSessionImpl();
+
+    std::vector<WT_EXT **> stack(WT_SKIP_MAXDEPTH, nullptr);
+
+    SECTION("remove multiple extents and verify all extents after each remove")
```

Comment:
Is there any reason to define a `SECTION` if there's only one test in the overall test case?

Reply:
Reasons to have a SECTION even if the TEST_CASE has only one SECTION:
* I imagine a muiti-level outline of tests: tag, file (maybe just tag is enough), TEST_CASE, SECTION, and test within the SECTION. The last level is needed if a SECTION contains multiple tests whether from a loop or otherwise.
* For consistency in how a TEST_CASE comment is written.  TEST_CASE gives the function, and SECTION describes what is tested. Without this a description of what is written might not be given.
* For ease of adding more SECTIONs to the TEST_CASE.
* If we eventually scan unit tests to find out what they test and construct a list for a test plan. No plans to do that.

Reply:
Thanks Tod those reasons mostly make sense to me can you elaborate on this one?

> The last level is needed if a SECTION contains multiple tests whether from a loop or otherwise.

Why would the last level be needed if that singular section had multiple tests within it? It isn't like it would execute the reconstruction of the test much like other sections do.

Reply:
The idea is to make a list of all tests that are run and what they do in sufficient detail so it is possible to check if anything is not tested that should be. With multiple subtests or loops there needs to be a statement of what each of them is the subtests or loop cycles is testing.  That is the lowest level.